File: backend/app/services/dispatch.py
"""AI 调度服务 - 根据任务类型推荐 AI 工具"""

from sqlalchemy.orm import Session
from app.models.ai_tag import AITag
from app.models.dispatch_record import DispatchRecord
from app.models.task import Task
from app.schemas.dispatch import AITagUpdate, DispatchRequest


# 调度规则：任务类型 -> 首选 AI, 备选 AI
DISPATCH_RULES = {
    "backend": {"primary": "Claude Code", "secondary": "Codex", "reason": "代码生成能力强"},
    "frontend": {"primary": "Claude Code", "secondary": "Codex", "reason": "前端代码生成"},
    "test": {"primary": "Codex", "secondary": "Claude Code", "reason": "测试代码生成"},
    "docs": {"primary": "Claude Code", "secondary": "GPT", "reason": "文档编写"},
    "design": {"primary": "GPT", "secondary": "Claude Code", "reason": "方案设计"},
    "debug": {"primary": "Claude Code", "secondary": "Codex", "reason": "调试修复"},
}


def get_ai_tags(db: Session, workspace_id: int) -> list[AITag]:
    """获取项目空间的 AI 标签列表"""
    return db.query(AITag).filter(AITag.workspace_id == workspace_id).all()


def get_ai_tag(db: Session, ai_tool: str, workspace_id: int) -> AITag | None:
    """获取单个 AI 标签"""
    return db.query(AITag).filter(
        AITag.ai_tool == ai_tool,
        AITag.workspace_id == workspace_id,
    ).first()


def update_ai_tag(db: Session, ai_tool: str, workspace_id: int, data: AITagUpdate) -> AITag:
    """更新 AI 标签"""
    tag = get_ai_tag(db, ai_tool, workspace_id)
    if not tag:
        tag = AITag(ai_tool=ai_tool, workspace_id=workspace_id)
        db.add(tag)

    if data.tags is not None:
        tag.tags = data.tags
    if data.stats is not None:
        tag.stats = data.stats

    db.commit()
    db.refresh(tag)
    return tag


def dispatch_task(db: Session, data: DispatchRequest) -> DispatchRecord:
    """调度任务到 AI 工具

    根据任务类型和 AI 标签推荐最合适的 AI 工具。
    """
    task = db.query(Task).filter(Task.id == data.task_id).first()
    if not task:
        raise ValueError("任务不存在")

    # 确定任务类型
    task_type = data.task_type or "backend"

    # 获取调度规则
    rule = DISPATCH_RULES.get(task_type, DISPATCH_RULES["backend"])

    # 确定 AI 工具
    ai_tool = data.preferred_ai or rule["primary"]

    # 生成执行提示词（结构化模板，隔离用户输入与系统指令）
    execution_prompt = (
        f"系统指令：请根据以下任务信息完成工作。\n"
        f"--- 任务信息开始 ---\n"
        f"标题: {task.title}\n"
        f"描述: {task.description or '无'}\n"
        f"类型: {task_type}\n"
        f"执行体: {task.executor_name or '未指定'}\n"
        f"--- 任务信息结束 ---"
    )

    # 创建调度记录
    record = DispatchRecord(
        task_id=data.task_id,
        ai_tool=ai_tool,
        dispatch_reason=rule["reason"],
        risk_level="medium",
        execution_prompt=execution_prompt,
    )
    db.add(record)
    db.commit()
    db.refresh(record)
    return record


def get_dispatch_records(db: Session, task_id: int | None = None) -> list[DispatchRecord]:
    """获取调度记录"""
    query = db.query(DispatchRecord)
    if task_id:
        query = query.filter(DispatchRecord.task_id == task_id)
    return query.order_by(DispatchRecord.created_at.desc()).all()

File: frontend/src/mocks/dispatch.ts
import type { AITag, DispatchResponse, DispatchRecord } from '@/types/dispatch'

const mockAITags: AITag[] = [
  {
    id: 1,
    ai_tool: 'Claude Code',
    workspace_id: 1,
    tags: { specialty: ['backend', 'frontend', 'docs'] },
    stats: { speed: 0.8, quality: 0.7, stability: 0.9 },
    created_at: '2026-05-22T00:00:00Z',
    updated_at: '2026-05-22T00:00:00Z',
  },
  {
    id: 2,
    ai_tool: 'Codex',
    workspace_id: 1,
    tags: { specialty: ['test', 'debug'] },
    stats: { speed: 0.6, quality: 0.8, stability: 0.7 },
    created_at: '2026-05-22T00:00:00Z',
    updated_at: '2026-05-22T00:00:00Z',
  },
  {
    id: 3,
    ai_tool: 'GPT',
    workspace_id: 1,
    tags: { specialty: ['design', 'docs'] },
    stats: { speed: 0.7, quality: 0.75, stability: 0.85 },
    created_at: '2026-05-22T00:00:00Z',
    updated_at: '2026-05-22T00:00:00Z',
  },
]

const mockRecords: DispatchRecord[] = [
  {
    id: 1,
    task_id: 1,
    ai_tool: 'Claude Code',
    dispatch_reason: '代码生成任务，推荐使用 Claude Code',
    risk_level: 'low',
    execution_prompt: '请实现用户登录功能，包含 JWT 认证',
    execution_result: '已完成 Login.vue 和 auth store 的实现',
    duration_seconds: 45,
    success: true,
    created_at: '2026-05-22T10:01:00Z',
  },
]

export const mockDispatchApi = {
  dispatch(data: { task_id: number; ai_tool?: string }): Promise<DispatchResponse> {
    return Promise.resolve({
      recommended_tool: data.ai_tool || 'Claude Code',
      reason: '基于任务类型分析，推荐使用 Claude Code 执行代码生成任务',
      risk_level: 'low',
      execution_prompt: `请执行以下任务（Task #${data.task_id}）：\n根据上下文信息完成代码实现`,
      alternatives: [
        { tool: 'Codex', reason: '备选方案，适合测试代码生成' },
        { tool: 'GPT', reason: '适合文档和方案设计' },
      ],
    })
  },
  getRecords(taskId?: number): Promise<DispatchRecord[]> {
    if (taskId) return Promise.resolve(mockRecords.filter((r) => r.task_id === taskId))
    return Promise.resolve(mockRecords)
  },
  getAITags(workspaceId?: number): Promise<AITag[]> {
    if (workspaceId) return Promise.resolve(mockAITags.filter((t) => t.workspace_id === workspaceId))
    return Promise.resolve(mockAITags)
  },
  updateAITag(aiTool: string, data: { tags?: Record<string, unknown>; stats?: Record<string, unknown> }): Promise<AITag> {
    const tag = mockAITags.find((t) => t.ai_tool === aiTool)
    if (!tag) return Promise.reject(new Error('AI Tag not found'))
    if (data.tags) tag.tags = { ...tag.tags, ...data.tags }
    if (data.stats) tag.stats = { ...tag.stats, ...data.stats }
    tag.updated_at = new Date().toISOString()
    return Promise.resolve(tag)
  },
}
